packages/wagmi/src/utils/defaultWagmiReactConfig.ts

```diff
@@ -1,62 +1,68 @@
-import { InjectedConnector } from 'wagmi/connectors/injected'
-import { WalletConnectConnector } from 'wagmi/connectors/walletConnect'
-import { publicProvider } from 'wagmi/providers/public'
-import { EIP6963Connector } from '../connectors/EIP6963Connector.js'
-import { EmailConnector } from '../connectors/EmailConnector.js'
-import type { ConfigOptions } from './defaultWagmiCoreConfig.js'
-import { walletConnectProvider } from './provider.js'
+import type { Config, CreateConnectorFn } from '@wagmi/core'
+import { type Chain } from 'viem/chains'
+import { createConfig } from '@wagmi/core'
+
+import { createClient, http } from 'viem'
+import { coinbaseWallet, walletConnect, injected } from '@wagmi/connectors'
+import { emailConnector } from '../connectors/EmailConnector.js'
+
+export interface ConfigOptions {
+  projectId: string
+  chains: Chain[]
+  metadata: {
+    name: string
+    description: string
+    url: string
+    icons: string[]
+    verifyUrl: string
+  }
+  enableInjected?: boolean
+  enableEIP6963?: boolean
+  enableCoinbase?: boolean
+  enableEmail?: boolean
+  enableWalletConnect?: boolean
+}
 
 export function defaultWagmiConfig({
   projectId,
   chains,
   metadata,
   enableInjected,
   enableCoinbase,
-  enableEIP6963,
+  enableWalletConnect,
   enableEmail,
-  enableWalletConnect
-}: ConfigOptions) {
-  const { publicClient } = configureChains(chains, [
-    walletConnectProvider({ projectId }),
-    publicProvider()
-  ])
-
-  const connectors: Connector[] = []
+  enableEIP6963
+}: ConfigOptions): Config {
+  const connectors: CreateConnectorFn[] = []
 
   // Enabled by default
   if (enableWalletConnect !== false) {
-    connectors.push(
-      new WalletConnectConnector({ chains, options: { projectId, showQrModal: false, metadata } })
-    )
+    connectors.push(walletConnect({ projectId, metadata, showQrModal: false }))
   }
 
   if (enableInjected !== false) {
-    connectors.push(new InjectedConnector({ chains, options: { shimDisconnect: true } }))
-  }
-
-  if (enableEIP6963 !== false) {
-    connectors.push(new EIP6963Connector({ chains }))
+    connectors.push(injected({ shimDisconnect: true }))
   }
 
   if (enableCoinbase !== false) {
-    connectors.push(
-      new CoinbaseWalletConnector({ chains, options: { appName: metadata?.name ?? 'Unknown' } })
-    )
+    connectors.push(coinbaseWallet({ appName: metadata?.name ?? 'Unknown' }))
   }
 
   // Dissabled by default
   if (enableEmail === true) {
-    connectors.push(new EmailConnector({ chains, options: { projectId } }))
+    connectors.push(emailConnector({ chains, options: { projectId } }))
   }
 
   return createConfig({
-    autoConnect: true,
+    chains: chains as [Chain, ...Chain[]],
+    client: ({ chain }) =>
+      createClient({
+        chain,
+        transport: http()
+      }),
     connectors,
-    publicClient
+    multiInjectedProviderDiscovery: enableEIP6963 !== false
```

Comment:
in order to support more features (like SSR) in our defaultWagmiConfig can we add a spread operator?

```suggestion
    multiInjectedProviderDiscovery: enableEIP6963 !== false,
    ...wagmiOptions
```